spec/models/odk/response_parser_spec.rb

```diff
@@ -160,27 +172,54 @@
         end
 
         context "with invalid data types" do
-          let(:xml_values) { ["9.6", "The quick brown fox", "foo", "bar"] }
+          let(:xml_values) { ["9.6", "The quick brown fox", "foo", ""] }
           let(:expected_values) { ["9.6", "The quick brown fox", 0.0, nil] }
 
           it "processes values correctly" do
             Odk::ResponseParser.new(response: response, files: files).populate_response
             expect_built_children(response.root_node, %w[Answer] * 4, form.c.map(&:id), expected_values)
           end
         end
+
+        context "with option from another mission" do
+          let(:other_mission) { create(:mission) }
+          let(:other_form) { create(:form, mission: other_mission, question_types: question_types) }
+          let(:other_option_code) { "on#{other_form.c[3].option_set.c[0].id}" }
+          let(:xml_values) { ["Quick", "The quick brown fox", "9.6", other_option_code] }
+
+          it "should error" do
+            expect do
+              Odk::ResponseParser.new(response: response, files: files).populate_response
+            end.to raise_error(SubmissionError, /unidentifiable option node/)
+          end
+        end
```

Comment:
Having thought about this more, I think we actually can strengthen all of this to say the option node has to be not just in the mission, but also in the option set for the question. The integrity model says there can be a hard fail if you remove an option from an option set, or if you change the option set.

Sorry for not realizing this sooner! Would you agree?

Reply:
[while pairing we agreed that either way is fine for this]